struct.go

```diff
@@ -244,7 +244,7 @@ func (s *Section) mapTo(val reflect.Value, isStrict bool) error {
 			continue
 		}
 
-		isAnonymous := tpField.Type.Kind() == reflect.Ptr && tpField.Anonymous
+		isAnonymous := tpField.Type.Kind() == reflect.Ptr && (tpField.Anonymous || tpField.Type.Elem().Kind() == reflect.Struct)
```

Comment:
The purpose of `isAnonymous` variable is to indicate if the field is anonymous, what you have edited might work at this time, but it is not what it meant to be.

You should consider modifying line 248.